### PR TITLE
ROGER-864:Automated Test Cases for Authorization

### PR DESCRIPTION
**Pull request by 04msambit**
Automated the Test Cases which were manually executed:

Sample Output:

1 - Test 1 : When no environment or machine name is provided.

sambits-mbp:test sambitmishra$ python marathon_auth_plugin_test.py
Please provide list of machines and retry.
Exiting...

2 - Test 2 : When a valid environment is provided:

sambits-mbp:test sambitmishra$ python marathon_auth_plugin_test.py localmesos01

Executing Tests for environemnt : localmesos01

Executing Test: User with Root Access:

Create Test: Pass
Read Test: Pass
Update Test: Pass
Delete Test: Pass

Executing Test: User with No Access:

Create Test: Pass
Read Test: Pass
Update Test: Pass
Delete Test: Pass

Executing Test: User with Dev Access:

Create Test: Pass
Read Test: Pass
Update Test: Pass
Delete Test: Pass

Executing Test: User with Shared Directory Access:

Create Test: Pass
Read Test: Pass
Update Test: Pass
Delete Test: Pass

Executing Test: CRUD on Unauthorized Environment:

Create Test: Not Allowed to Create -- Result: Pass
Read Test: Not Allowed to Read -- Result: Pass
Update Test: Not Allowed to Update -- Result: Pass
Delete Test: Not Allowed to Delete -- Result: Pass

3 Test 3 : When an invalid environment is provided:

python marathon_auth_plugin_test.py localmesos01 localmesos02

Executing Tests for environemnt : localmesos01

Executing Test: User with Root Access:

Create Test: Pass
Read Test: Pass
Update Test: Pass
Delete Test: Pass

Executing Test: User with No Access:

Create Test: Pass
Read Test: Pass
Update Test: Pass
Delete Test: Pass

Executing Test: User with Dev Access:

Create Test: Pass
Read Test: Pass
Update Test: Pass
Delete Test: Pass

Executing Test: User with Shared Directory Access:

Create Test: Pass
Read Test: Pass
Update Test: Pass
Delete Test: Pass

Executing Test: CRUD on Unauthorized Environment:

Create Test: Not Allowed to Create -- Result: Pass
Read Test: Not Allowed to Read -- Result: Pass
Update Test: Not Allowed to Update -- Result: Pass
Delete Test: Not Allowed to Delete -- Result: Pass

Environment localmesos02: Unreachable

@vmahedia Please review. 
NOTE : For the case where we negate all user-permission and run these same tests. All test cases fail.

### CODE REVIEW

File: javaauth/test/marathon_auth_plugin_test.py [hunk 1]
```diff
@@ -0,0 +1,422 @@
...
+    except:
+        print("Failed to store original file from remote server")
+
+def scp_to_remote_location(filepath):
+    try:
+        user = os.getlogin()
+        remote_path = "/data/marathon/plugins/user-permissions.json"
+        hostname1 = user+"@daldevmesoszk01"
+        hostname2 = user+"@daldevmesoszk02"
+        hostname3 = user+"@daldevmesoszk03"
```
Comment (vmahedia) [inline javaauth/test/marathon_auth_plugin_test.py]: why are they hardcoded?

File: javaauth/test/marathon_auth_plugin_test.py [hunk 2]
```diff
@@ -0,0 +1,422 @@
...
+def delete_data(url,headers,payload):
+    url = url+payload['id']
+    result = requests.delete(url, data=json.dumps(payload), headers=headers)
+    return (result.status_code)
+
+def store_original_remote_file():
+    try:
+        user = os.getlogin()
+        remote_path = "/data/marathon/plugins/user-permissions.json"
+        hostname1 = user+"@daldevmesoszk01"
```
Comment (vmahedia) [inline javaauth/test/marathon_auth_plugin_test.py]: hardcoded machine name???

Comment (vmahedia) [inline javaauth/test/marathon_auth_plugin_test.py]: I believe what you are doing is - saving the existing `user-permissions.json` file before you SCP your test `permissions.json` which is correct thing to do, just use the hostname supplied on the command line, This same tests will also be ran on other environment as well.

File: javaauth/test/marathon_auth_plugin_test.py (hunk 1 above)
Comment (04msambit) [inline javaauth/test/marathon_auth_plugin_test.py]: Removed the hardcoded values. The machine names are now passed as an parameter in the command line.

File: javaauth/test/marathon_auth_plugin_test.py (hunk 2 above)
Comment (04msambit) [inline javaauth/test/marathon_auth_plugin_test.py]: Removed the hardcoded values. Also we can provide a flag = "shared-storage" , if we want to copy the permissions file to only one location.

**Comment by vmahedia** [comment]
LGTM

**Pull request merged**